chunks/generative_models.ipynb

```diff
@@ -0,0 +1,397 @@
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Sampling in practice\n",
+    "\n",
+    "* In general, sampling from a PDF is difficult\n",
+    "\n",
+    "* For certain standard distributions, however, there are fast algorithms"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([0, 2, 4, 7, 0, 3, 2, 3, 3, 1])"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import scipy.stats\n",
+    "\n",
+    "P = scipy.stats.poisson(mu=3)\n",
+    "P.rvs(size=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Sampling in practice\n",
+    "\n",
+    "* [numpy.random]() and [scipy.stats]() are two useful libraries for drawing samples from PDFs\n",
+    "\n",
+    "* You may have encountered some of these routines as \"random number generators\"\n",
+    "\n",
+    "* Sampling from a PDF means generating random numbers from that distribution"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Choosing the input model parameters\n",
+    "\n",
+    "* When testing, we often want to assert a set of input model parameters $\\theta$, and then see what they produce\n",
+    "\n",
+    "* Testing at large scale might involve generating many datasets, with different inputs. In this case we might want to sample from a *plausible distribution of input parameters*\n",
+    "\n",
+    "* In practice: choose a particular standard probability distribution for $\\theta$ and sample from it"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Deterministic relationships\n",
+    "\n",
+    "* Often, our model provides a deterministic relationship between parameters: if you know $\\theta$, then you know $\\mu$\n",
+    "\n",
+    "* In our X-ray image case, $\\theta$ could be the set of parameters that describe the gas temperature and density profiles of a spehrically symmetric cluster of galaxies with known centroid position\n",
+    "\n",
+    "* $\\mu(\\theta)$ would then be some complicated function that took the cluster parameters $\\theta$ as input, and predicted the expected counts at any pixel position."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Probabilistic Graphical Models\n",
+    "\n",
+    "* This procedure for simulating a mock X-ray image dataset can be usefully illustrated with a _directed acyclic graph_ called a \"Probabilistic Graphical Model\"\n",
```

Comment:
I was thinking of making this an active comprehension exercise - Ill make one and see what it looks like